src/main/scala/no/ndla/draftapi/integration/TaxonomyApiClient.scala

```diff
@@ -0,0 +1,148 @@
+                                   updateTranslationFunc: (String, String, String) => Try[Translation]) = {
+      val tries = titles.map(t => updateTranslationFunc(id, t.language, t.title))
+      Traverse[List].sequence(tries.toList)
+    }
+
+    private def updateResourceTitleAndTranslations(res: Resource,
+                                                   defaultTitle: ArticleTitle,
+                                                   titles: Seq[ArticleTitle]) = {
+      val updateTranslationsFunc = updateTranslations(res.id, _: Seq[ArticleTitle], updateResourceTranslation)
+      updateTitleAndTranslations(res, defaultTitle, titles, updateResource, updateTranslationsFunc)
+    }
+
+    private def updateTopicTitleAndTranslations(top: Topic, defaultTitle: ArticleTitle, titles: Seq[ArticleTitle]) = {
+      val updateTranslationsFunc = updateTranslations(top.id, _: Seq[ArticleTitle], updateTopicTranslation)
+      updateTitleAndTranslations(top, defaultTitle, titles, updateTopic, updateTranslationsFunc)
+    }
+
+    private[integration] def updateResourceTranslation(resourceId: String, lang: String, name: String) =
+      putRaw(s"$TaxonomyApiEndpoint/resources/$resourceId/translations/$lang", Translation(name))
+
+    private[integration] def updateTopicTranslation(topicId: String, lang: String, name: String) =
+      putRaw(s"$TaxonomyApiEndpoint/topics/$topicId/translations/$lang", Translation(name))
+
+    private[integration] def updateResource(resource: Resource)(implicit formats: Formats) =
+      putRaw[Resource](s"$TaxonomyApiEndpoint/resources/${resource.id}", resource)
+
+    private[integration] def updateTopic(topic: Topic)(implicit formats: Formats) =
+      putRaw[Topic](s"$TaxonomyApiEndpoint/topics/${topic.id}", topic)
+
+    private def get[A](url: String, params: (String, String)*)(implicit mf: Manifest[A]): Try[A] =
+      ndlaClient.fetchWithForwardedAuth[A](Http(url).timeout(taxonomyTimeout, taxonomyTimeout).params(params))
+
+    def queryResource(articleId: Long): Try[List[Resource]] =
+      get[List[Resource]](s"$TaxonomyApiEndpoint/queries/resources", "contentURI" -> s"urn:article:$articleId")
+
+    def queryTopic(articleId: Long): Try[List[Topic]] =
+      get[List[Topic]](s"$TaxonomyApiEndpoint/queries/topics", "contentURI" -> s"urn:article:$articleId")
+
+    private[integration] def putRaw[B <: AnyRef](url: String, data: B, params: (String, String)*)(
+        implicit formats: org.json4s.Formats): Try[B] = {
+      logger.info(s"Doing call to $url")
+      ndlaClient.fetchRawWithForwardedAuth(
+        Http(url)
+          .put(write(data))
+          .timeout(taxonomyTimeout, taxonomyTimeout)
+          .header("content-type", "application/json")
+          .params(params)
+      ) match {
+        case Success(_)  => Success(data)
+        case Failure(ex) => Failure(ex)
+      }
+    }
+  }
+}
+
+trait Taxonomy[E <: Taxonomy[E]] {
+  self: E =>
```

Comment:
Hva er denne til?

Reply:
Ser ikke ut som den er nødvendig, duden på stack overflow skrev: 
`
In addition, I would also recommend you to use a self typing to enforce the return type of your withId() method.
`
https://stackoverflow.com/questions/15441589/scala-copy-case-class-with-generic-type

Skal jeg fjerne den? :hamster:

Reply:
Hmmm, jeg skjønner ikke helt hvordan self-typing på den måten kan enforce return typen på `withId`.. I følge [scala-docen](https://docs.scala-lang.org/tour/self-types.html) er self typing bare for å definere hvilke andre traits som må mixes inn.
Hvis ingen av oss skjønner poenget med den foreslår jeg å fjerne det :)